zospy/zpcore.py

```diff
@@ -414,11 +443,33 @@ def create_new_application(self) -> bool:
 
         self.Application = self.Connection.CreateNewApplication()
 
-        if self.Application.IsValidLicenseForAPI:
-            return True
-        else:
+        if not self.Application.IsValidLicenseForAPI:
             logger.critical("OpticStudio Licence is not valid for API, connection not established")
+            raise ConnectionRefusedError("OpticStudio Licence is not valid for API, connection not established")
             return False
+        else:
+            if not return_primary_system:
+                return True
+            else:
+                return self.get_primary_system()
+ 
+    def connect_as_standalone(self, return_primary_system: bool = False) -> bool | OpticStudioSystem:
```

Comment:
The suggestions for the docstring of `connect_as_extension` go for this one as well.